agent/asm/asm.go

```diff
@@ -86,3 +86,19 @@ func extractASMValue(out *secretsmanager.GetSecretValueOutput) (docker.AuthConfi
 
 	return dac, nil
 }
+
+// GetSecretFromASM makes the api call to the AWS Secrets Manager service to
+// retrieve the secret value
+func GetSecretFromASM(secretID string, client secretsmanageriface.SecretsManagerAPI) (string, error) {
+	in := &secretsmanager.GetSecretValueInput{
+		SecretId: aws.String(secretID),
+	}
+
+	out, err := client.GetSecretValue(in)
```

Comment:
Does this have retries? Is there a possibility that customer would hit throttle errors here?

Reply:
AWS SDK by default implements retry for their API calls. Customer may hit throttle error, however, they should be able to contact support to increase their limit.